core/shared/src/main/scala/scalaz/zio/internal/Stack.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2018 - 2019 John A. De Goes. All rights reserved.
+package scalaz.zio.internal
+
+/**
+ * A very fast, growable/shrinkable, mutable stack.
+ */
+final class Stack[A <: AnyRef]() {
```

Comment:
not ArrayDeque?

Reply:
no chunk ?
FYI:https://github.com/real-logic/agrona/blob/master/agrona/src/main/java/org/agrona/concurrent/OneToOneConcurrentArrayQueue.java
https://github.com/JCTools/JCTools/blob/master/jctools-core/src/main/java/org/jctools/queues/SpscChunkedArrayQueue.java

Reply:
No third-party dependencies and this version is really quite fast.